clarity_epp/export/illumina.py

```diff
@@ -146,148 +199,93 @@ def get_project(projects, urgent=False):
+        samplesheet_samples.append(get_samplesheet_samples(sample_artifacts, process, index_2_orientation))
+
+    # Create SampleSheet
+    sample_sheet = []
+
+    # Header
+    sample_sheet.append('[Header]')
+    sample_sheet.append('FileFormatVersion,2')
+    sample_sheet.append('RunName,{0}'.format(process.udf['Experiment Name']))
+
+    # Reads
+    sample_sheet.append('[Reads]')
+    sample_sheet.append('Read1Cycles,{0}'.format(process.udf['Read 1 Cycles']))
+    sample_sheet.append('Read2Cycles,{0}'.format(process.udf['Read 2 Cycles']))
+    sample_sheet.append('Index1Cycles,{0}'.format(process.udf['Index Read 1']))
+    sample_sheet.append('Index2Cycles,{0}'.format(process.udf['Index Read 2']))
+
+    # BCLConvert_Settings
+    sample_sheet.append('[BCLConvert_Settings]')
+    sample_sheet.append('AdapterRead1,{0}'.format(process.udf['Adapter']))
+    sample_sheet.append('AdapterRead2,{0}'.format(process.udf['Adapter Read 2']))
+    sample_sheet.append('FindAdaptersWithIndels,true')
+    sample_sheet.append('BarcodeMismatchesIndex1,0')
+    sample_sheet.append('BarcodeMismatchesIndex2,0')
+
+    # BCLConvert_Data
+    sample_sheet.append('[BCLConvert_Data]')
+    if len(samplesheet_samples) == 1:  # All samples on all lanes
+        lane = 0
+        sample_sheet.append('Sample_ID,index,index2,OverrideCycles,Sample_Project')
+        for sample in samplesheet_samples[lane]:
+            sample_sheet.append(
+                '{sample_name},{index_1},{index_2},{override_cycles},{project}'.format(
+                    sample_name=sample,
+                    index_1=samplesheet_samples[lane][sample]['index_1'],
+                    index_2=samplesheet_samples[lane][sample]['index_2'],
+                    override_cycles=samplesheet_samples[lane][sample]['override_cycles'],
+                    project=samplesheet_samples[lane][sample]['project']
                 )
-                custom_settings = 'TrimUMI,1\n'
-
-            custom_settings = '{settings}OverrideCycles,{override_cycles}\n'.format(
-                settings=custom_settings,
-                override_cycles=';'.join(override_cycles)
             )
+    else:  # Samples divided over lanes
+        sample_sheet.append('Lane,Sample_ID,index,index2,OverrideCycles,Sample_Project')
+        for lane, lane_samples in enumerate(samplesheet_samples):
+            for sample in lane_samples:
+                sample_sheet.append(
+                    '{lane},{sample_name},{index_1},{index_2},{override_cycles},{project}'.format(
+                        lane=lane+1,
+                        sample_name=sample,
+                        index_1=samplesheet_samples[lane][sample]['index_1'],
+                        index_2=samplesheet_samples[lane][sample]['index_2'],
+                        override_cycles=samplesheet_samples[lane][sample]['override_cycles'],
+                        project=samplesheet_samples[lane][sample]['project']
+                    )
+                )
```

Comment:
Very similar code, differences:
- additional column 'Lane' and thus rowvalue lane.
- different object (lane_samples, samplesheet_samples).

Is it possible to combine somehow?

Reply:
You are right, to similar. I think I found a nice solution :)